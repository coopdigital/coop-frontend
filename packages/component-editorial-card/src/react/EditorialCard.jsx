import React from "react";
import PropTypes from "prop-types";
// import '../css/EditorialCard.pcss';

const EditorialCard = ({ title, text, link, type, testId }) => {
  const tagAttributes = {
    className: "coop coop-c-editorialcard coop-c-editorialcard--horizontal",
    role: type === "warn" || type === "error" ? "alert" : "status",
  };

  return (
    <div
      data-testid={testId}
      className={tagAttributes.className}
      role={tagAttributes.role}
    >
      <div className="coop-c-editorialcard__inner ">
        <h3 className="coop-c-editorialcard__title coop-c-notification__heading">{title}</h3>
        <p className="coop-c-notification__p">
          {link ? (
            <a href={link} className="coop-c-notification__link">
              {text}
            </a>
          ) : (
            text
          )}
        </p>
      </div>
    </div>
  );
};

EditorialCard.defaultProps = {
  type: "info",
};

EditorialCard.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  link: PropTypes.string,
  type: PropTypes.oneOf(["info", "warn", "error", "success"]),
  testId: PropTypes.string,
};

export default EditorialCard;
