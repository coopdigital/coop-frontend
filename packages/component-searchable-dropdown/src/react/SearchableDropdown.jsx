import React, { useRef, useEffect } from "react";
import SearchableDropdownLibrary from "../lib/SearchableDropdownLibrary.js";

const SearchableDropdown = () => {
  const comboboxRef = useRef();

  useEffect(() => {
    new SearchableDropdownLibrary(comboboxRef.current).setup();
  }, []);

  return (
    <div ref={comboboxRef}>
      <div data-combobox>
        <label htmlFor="combobox">How you know them</label>
        <input
          name="combobox"
          className="coop-form__input"
          type="text"
          placeholder="Select option"
          data-input
          data-testid="combobox-input"
          role="combobox"
          aria-controls="comboboxvalues"
          aria-autocomplete="list"
          aria-expanded="false"
        />
        <div
          style={{ display: "none" }}
          data-dropdown
          data-testid="combobox-dropdown"
        >
          <ul id="comboboxvalues" role="listbox" data-list>
            {/* These valuse would be mapped from a data prop */}
            <li role="option" aria-selected="false" data-option>
              Partner
            </li>
            <li role="option" aria-selected="false" data-option>
              Spouse
            </li>
            <li role="option" aria-selected="false" data-option>
              Parent
            </li>
            <li role="option" aria-selected="false" data-option>
              Child
            </li>
            <li role="option" aria-selected="false" data-option>
              Brother
            </li>
            <li role="option" aria-selected="false" data-option>
              Grandchild
            </li>
            <li role="option" aria-selected="false" data-option>
              Father
            </li>
          </ul>
        </div>
      </div>
    </div>
  );
};

export default SearchableDropdown;
