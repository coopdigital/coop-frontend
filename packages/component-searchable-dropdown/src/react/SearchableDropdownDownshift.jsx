/* eslint-disable indent */
import React, { useState } from 'react';
import { useCombobox } from 'downshift';
import PropTypes from 'prop-types';
import { matchSorter } from 'match-sorter';

const SearchableDropdown = ({
  className,
  compact,
  id,
  label,
  noResults,
  onSelect,
  options,
  placeholder,
  style,
}) => {
  const [inputItems, setInputItems] = useState(options);
  const sizeVariant = compact ? 'compact' : '';
  const noResultsText = noResults || 'No relevant options';
  const placeholderValue = placeholder || null;

  const {
    isOpen,
    closeMenu,
    getToggleButtonProps,
    getLabelProps,
    getMenuProps,
    getInputProps,
    getComboboxProps,
    openMenu,
    getItemProps,
  } = useCombobox({
    id,
    items: inputItems,
    onInputValueChange: ({ inputValue }) => {
      setInputItems(matchSorter(options, inputValue));
    },
    onSelectedItemChange: ({ selectedItem: selected }) => {
      if (typeof onSelect === 'function') onSelect(selected);
      closeMenu();
    },
  });

  return (
    <div className={`coop-c-combobox ${sizeVariant} ${className}`} style={style}>
      <label {...getLabelProps()} htmlFor={`${id}-input`}>
        {label}
      </label>
      <div {...getComboboxProps()}>
        <input {...getInputProps({ onFocus: openMenu })} placeholder={placeholderValue} />
        <button type="button" {...getToggleButtonProps()} aria-label="toggle menu">
          <svg xmlns="http://www.w3.org/2000/svg" height="24" width="24">
            <path d="m12 15.375-6-6 1.4-1.4 4.6 4.6 4.6-4.6 1.4 1.4Z" />
          </svg>
        </button>
      </div>
      <ul {...getMenuProps()}>
        {isOpen && inputItems.length
          ? inputItems.map((item, index) => (
              <li {...getItemProps({ item, index })} key={`${item}${index}`}>
                {item}
              </li>
            ))
          : null}
        {isOpen && !inputItems.length ? (
          <li className="coop-c-combobox--null">{noResultsText}</li>
        ) : null}
      </ul>
    </div>
  );
};

export { SearchableDropdown };

SearchableDropdown.propTypes = {
  className: PropTypes.string,
  compact: PropTypes.bool,
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  noResults: PropTypes.string,
  onSelect: PropTypes.func,
  options: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.number, PropTypes.string])).isRequired,
  placeholder: PropTypes.string,
  style: PropTypes.object,
};
