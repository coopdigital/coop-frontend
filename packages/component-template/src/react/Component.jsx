import React from "react";

const Component = () => <div class="coop-c-component">Example HTML</div>;

export default Component;
