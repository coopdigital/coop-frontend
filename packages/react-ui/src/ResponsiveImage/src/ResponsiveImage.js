import React from "react";
import PropTypes from "prop-types";
import shortid from "shortid";

const generateSourceMap = (sources) =>
  sources.map((source) => {
    const { type, media, srcSet } = source;
    const srcSets = Object.entries(srcSet).reduce(
      (acc, [descriptor, imgSrc]) => {
        acc.push(`${imgSrc} ${descriptor}`);
        return acc;
      },
      []
    );

    const sourceProps = { type, media };
    return (
      <source
        data-testid="test-picture-source"
        key={shortid.generate()}
        srcSet={srcSets.join(", ")}
        {...sourceProps}
      />
    );
  });

const ResponsiveImage = ({ src, alt, width, height, sources }) => {
  return (
    <figure className="coop-c-image">
      <picture data-testid="test-picture-el" className="coop-c-image__media">
        {generateSourceMap(sources)}
        <img src={src} alt={alt} width={width} height={height} />
      </picture>
    </figure>
  );
};

ResponsiveImage.defaultProps = {
  width: null,
  height: null,
  sources: [],
};

ResponsiveImage.propTypes = {
  /** src - (required) a url to the image you are sourcing. */
  src: PropTypes.string.isRequired,
  /** alt - (required) needed to give a meaningful description to screenreader users. */
  alt: PropTypes.string.isRequired,
  /** width - can set the width of the image in pixels */
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** height - can set the height of the image in pixels */
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** source - takes an array of objects with the following properties: `type`, `media` and `srcSet` */
  sources: PropTypes.arrayOf(PropTypes.object),
};

export default ResponsiveImage;
