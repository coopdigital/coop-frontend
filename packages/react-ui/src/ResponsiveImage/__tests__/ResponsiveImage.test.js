/**
 * @jest-environment jsdom
 */
import React from "react";
import { render, cleanup } from '@testing-library/react';

import ResponsiveImage from "../src/index";

afterEach(cleanup);

describe("ResponsiveImage", () => {
  it("should render correctly", () => {
    const wrapper = render(<ResponsiveImage src="src" alt="alt" />);
    expect(() => wrapper.unmount()).not.toThrow();

    const { asFragment } = render(<ResponsiveImage src="src" alt="alt" />);
    const imageFragment = asFragment();
    expect(imageFragment).toMatchSnapshot();
  });

  it("should allow width and height attributes on default src", () => {
    const {asFragment} = render(<ResponsiveImage src="src" alt="alt" width="120" height="120" />);
    const imageFragmentWidthHeight = asFragment();
    expect(imageFragmentWidthHeight).toMatchSnapshot();
  });

  it("should render sources", () => {
    const sourcesTest = [
      {
        type: "image/webp",
        media: "(min-width: 1200)",
        srcSet: {
          "1x":
            "//images.ctfassets.net/bffxiku554r1/7XDwxNP8ioupeMdJZxNS8/d23d2b802d6be0ecbcb7abd1ff33b159/coop-supporting-local-communities.png?fm=webp&amp;q=60&amp;w=600&amp;h=338",
          "2x":
            "//images.ctfassets.net/bffxiku554r1/7XDwxNP8ioupeMdJZxNS8/d23d2b802d6be0ecbcb7abd1ff33b159/coop-supporting-local-communities.png?fm=webp&amp;q=60&amp;w=600&amp;h=338",
        },
      },
      {
        type: "image/webp",
        media: "(min-width: 600)",
        srcSet: {
          "1x":
            "//images.ctfassets.net/bffxiku554r1/7XDwxNP8ioupeMdJZxNS8/d23d2b802d6be0ecbcb7abd1ff33b159/coop-supporting-local-communities.png?fm=webp&amp;q=60&amp;w=600&amp;h=338",
          "2x":
            "//images.ctfassets.net/bffxiku554r1/7XDwxNP8ioupeMdJZxNS8/d23d2b802d6be0ecbcb7abd1ff33b159/coop-supporting-local-communities.png?fm=webp&amp;q=60&amp;w=600&amp;h=338",
        },
      },
    ];
    const {asFragment} = render(<ResponsiveImage src="src" alt="alt" sources={sourcesTest} />);
    const imageFragmentWithSourceSet = asFragment();
    expect(imageFragmentWithSourceSet).toMatchSnapshot();
  });
});
