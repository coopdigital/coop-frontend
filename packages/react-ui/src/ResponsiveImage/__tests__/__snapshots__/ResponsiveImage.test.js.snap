// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveImage should allow width and height attributes on default src 1`] = `
<DocumentFragment>
  <figure
    class="coop-c-image"
  >
    <picture
      class="coop-c-image__media"
    >
      <img
        alt="alt"
        height="120"
        src="src"
        width="120"
      />
    </picture>
  </figure>
</DocumentFragment>
`;

exports[`ResponsiveImage should render correctly 1`] = `
<DocumentFragment>
  <figure
    class="coop-c-image"
  >
    <picture
      class="coop-c-image__media"
    >
      <img
        alt="alt"
        src="src"
      />
    </picture>
  </figure>
</DocumentFragment>
`;

exports[`ResponsiveImage should render sources 1`] = `
<DocumentFragment>
  <figure
    class="coop-c-image"
  >
    <picture
      class="coop-c-image__media"
    >
      <source
        media="(min-width: 1200)"
        srcset="//images.ctfassets.net/bffxiku554r1/7XDwxNP8ioupeMdJZxNS8/d23d2b802d6be0ecbcb7abd1ff33b159/coop-supporting-local-communities.png?fm=webp&amp;q=60&amp;w=600&amp;h=338 1x, //images.ctfassets.net/bffxiku554r1/7XDwxNP8ioupeMdJZxNS8/d23d2b802d6be0ecbcb7abd1ff33b159/coop-supporting-local-communities.png?fm=webp&amp;q=60&amp;w=600&amp;h=338 2x"
        type="image/webp"
      />
      <source
        media="(min-width: 600)"
        srcset="//images.ctfassets.net/bffxiku554r1/7XDwxNP8ioupeMdJZxNS8/d23d2b802d6be0ecbcb7abd1ff33b159/coop-supporting-local-communities.png?fm=webp&amp;q=60&amp;w=600&amp;h=338 1x, //images.ctfassets.net/bffxiku554r1/7XDwxNP8ioupeMdJZxNS8/d23d2b802d6be0ecbcb7abd1ff33b159/coop-supporting-local-communities.png?fm=webp&amp;q=60&amp;w=600&amp;h=338 2x"
        type="image/webp"
      />
      <img
        alt="alt"
        src="src"
      />
    </picture>
  </figure>
</DocumentFragment>
`;
